piccolo/query/mixins.py

```diff
@@ -784,3 +784,51 @@ def on_conflict(
                 target=target, action=action_, values=values, where=where
             )
         )
+
+
+@dataclass
+class ForUpdate:
+    __slots__ = ("nowait", "skip_locked", "of")
+
+    nowait: bool
+    skip_locked: bool
+    of: tuple[Table]
+
+    def __post_init__(self):
+        if not isinstance(self.nowait, bool):
+            raise TypeError("nowait must be an integer")
+        if not isinstance(self.skip_locked, bool):
+            raise TypeError("skip_locked must be an integer")
+        if not isinstance(self.of, tuple) or not all(
+            hasattr(x, "_meta") for x in self.of
+        ):
+            raise TypeError("of must be an tuple of Table")
+        if self.nowait and self.skip_locked:
+            raise TypeError(
+                "The nowait option cannot be used with skip_locked"
+            )
+
+    @property
+    def querystring(self) -> QueryString:
+        sql = " FOR UPDATE"
+        if self.of:
+            tables = ", ".join(x._meta.tablename for x in self.of)
```

Comment:
shouldn't the args be escaped?

Reply:
All other code assumes that `table._meta.tablename` is safe. An extra sanity check can be done in `piccolo.table.Table.__init_subclass__`, but it is not part of this PR